sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -105,26 +105,60 @@ object GpuParquetFileFormat {
     }
   }
 
-  def parquetWriterOptionsFromSchema[T <: NestedBuilder[_, _], V <: ParquetColumnWriterOptions]
-  (builder: ParquetColumnWriterOptions.NestedBuilder[T, V],
-   schema: StructType, writeInt96: Boolean): T = {
+  def getParquetColumnWriterOption(
+      name: String,
+      dataType: DataType,
+      writeInt96: Boolean,
+      nullable: Boolean): ParquetColumnWriterOptions = {
+    dataType match {
+      case dt: DecimalType =>
+        // key can never be null
+        new ParquetColumnWriterOptions(name, writeInt96, dt.precision, nullable)
+      case TimestampType =>
+        new ParquetColumnWriterOptions(name, writeInt96, 0, nullable)
+      case s: StructType =>
+        parquetWriterOptionsFromSchema(structBuilder(name, nullable), s, writeInt96).build()
+      case a: ArrayType =>
+        parquetWriterOptionsFromSchema(listBuilder(name),
+          StructType(Array(StructField(name, a.elementType, nullable))), writeInt96).build()
+      case m: MapType =>
+        mapColumn(name,
+          getParquetColumnWriterOption("key", m.keyType, writeInt96, false),
+          getParquetColumnWriterOption("value", m.valueType, writeInt96, nullable))
+      case _ =>
+        new ParquetColumnWriterOptions(name, nullable)
+    }
+  }
+
+  def parquetWriterOptionsFromSchema[T <: NestedBuilder[_, _], V <: ParquetColumnWriterOptions](
+      builder: ParquetColumnWriterOptions.NestedBuilder[T, V],
+      schema: StructType,
+      writeInt96: Boolean,
+      nullable: Boolean = true): T = {
     // TODO once https://github.com/rapidsai/cudf/issues/7654 is fixed go back to actually
     // setting if the output is nullable or not everywhere we have hard-coded nullable=true
```

Comment:
Has this been addressed?  Looks like the code is removing the hardcoded `true` nullable parameters.

Reply:
It has not been addressed. Cudf does not want to do it because of performance :). So it is unlikely to ever be addressed.